2020/day3.clj

```diff
@@ -44,31 +51,45 @@
         ...
         ".#..#...#.#"]
 
-  (take-while neg? [-2 0 1])
-  #_=> (-2)
-
-  (take 3 (map #(next-location [0 0] [(* 3 %) (* 1 %)])
-               (range)))
+  (take 3 (locations-s [3 1]))
   #_=> ([0 0] [3 1] [6 2])
 
-  (locations-s 3 [3 1])
-  #_=> ([0 0] [3 1] [6 2])
+  (take-while neg? [-2 0 1])
+  #_=> (-2)
 
-  (mod 10 5)
-  #_=> 0
+  (valid-locations (count sample-input) (take 5 (locations-s [3 1])))
+  #_=> ([0 0] [3 1] [6 2] [9 3] [12 4])
 
   (get-in sample-input [0 2])
   #_=> \#
 
   (get-in sample-input [0 0])
   #_=> \.
 
+  (char-at-location sample-input [0 0])
+  #_=> \.
+
+  (char-at-location sample-input [2 0])
+  #_=> \#
+
   (tree? \.)
   #_=> false
 
   (tree? \#)
   #_=> true
 
+  (def valid-locations-sample (valid-locations (count sample-input) (locations-s [3 1])))
+  #_=> #'advent-of-code.2020.day3/valid-locations-sample
+
+  (->> valid-locations-sample
+       (map (fn [loc] (char-at-location sample-input loc))))
```

Comment:
overly verbose, this is just a `partial` application of the function

Reply:
> overly verbose, this is just a `partial` application of the function

Ok, I've already deleted it.